contributed_definitions/nyaml/NXapm.yaml

```diff
@@ -1,538 +1,371 @@
-# IFES APT TC APT HDF5 v1 verifier
-# NEW ISSUE: possible offspring application definitions:
-# NXapm_opt/pl would be possible, as soon as NXluminescence by Carola Emminger and Florian Dobner is ready whereby
-# then there would be two subentries one for an NXapm and one for NXphotoluminesence to capture the photonic atom probe of Rouen/GPM
-# and finally if there were an NXapm_afm for atomic force microscopy the IMEC AFM/APT experiments could be stored with an NXapm_afm application definition
-# with NXapm and NXafm being respective subentries of the appdef but as NXapm_afm is a step-by-step approach one would have to release the constraint
-# that only a single measurement can be performed.
-# NXasat which could just take two subentries NXem and NXapm
-# NXasat should be a fuse of NXapm and NXem within an NXentry with NXsubentry, in this way we can combine the strength of both application definitions
-# to describe also the upcoming technique of atomic scale analytical tomography https://doi.org/10.1017/9781316677292
+  p: |
+    Number of pulses collected in between start_time and end_time resolved by an
+    instance of :ref:`NXevent_data_apm`. If this is not defined, p is the number of
+    ions included in the reconstructed volume if the application definition is used
+    to store results of an already reconstructed datasets.
+  n: |
+    Number of ions spatially filtered from results of the hit_finding algorithm
+    from which an instance of a reconstructed volume has been generated.
+    These ions get new identifier assigned in the process (the so-called
+    evaporation_identifier). This identifier must not be confused with
+    the pulse_identifier.
 type: group
 NXapm(NXobject):
   (NXentry):
-    exists: ['min', '1', 'max', 'unbounded']
-    \@version:
-      doc: |
-        An at least as strong as SHA256 hashvalue of the file
-        that specifies the application definition.
-    
-    # enumeration: [sha_256_hash]
-    definition:
+    exists: [min, 1, max, infty]
+    definition(NX_CHAR):
+      \@version(NX_CHAR):
+        enumeration: [NXapm]
+    profiling(NXcs_profiling):
       doc: |
-        NeXus NXDL schema to which this file conforms.
-      enumeration: [NXapm]
-    experiment_identifier:
-      doc: |
-        Ideally, a (globally) unique persistent identifier
-        for referring to this experiment.
-        
-        The identifier is usually defined/issued by the facility, laboratory,
-        or the principle investigator. The identifier enables to link
-        experiments to e.g. proposals.
-    experiment_description:
-      exists: optional
+        The configuration of the I/O writer software (e.g. `pynxtools <https://github.com/FAIRmat-NFDI/pynxtools>`_)
+        which was used to generate this NeXus file instance.
+      # command_line_call(NX_CHAR):
+      (NXprogram):  # understood by default as PROGRAM(NXprogram), i.e. program1, program2, ...
```

Comment:
This NXprogram or NXsoftware irrespective its name does not collide with NXroot.
In my opinion the approach taken in NXroot could anyhow be improved because the execution environment is not documented but that is a different topic.

As far as I understand the main difference between software and program is that the software is more a set of programs. A tool like AP Suite has an own semantic version plus 100  of dependencies. Ideally I would like to allow people to store all these 100 + 1. So program1 to program100 seems reasonable to me like as well software1 software100 however one could then also argue why not software/program1 software/program100 I am really undecided. Therefore, I suggest to merge this right now, make an own PR where this can be changed all the way. Also others should make a statement.